File: Python/compareVersion.py
def compareVersion(version1, version2):
    """
    :type version1: str
    :type version2: str
    :rtype: int
    """
    v1 = [int(x) for x in version1.split('.')]
    v2 = [int(x) for x in version2.split('.')]
    l1,l2=len(v1),len(v2)
    v1 += [0]*(l2-l1)
    v2 += [0]*(l1-l2)
    if v1 > v2:
        return 1
    elif v2 > v1:
        return -1
    else:
        return 0
    
  
 

#print compareVersion('01','1')
#print compareVersion('1','1.1')
print compareVersion('1.0.0','1')
    
        

File: JavaScript/compareVersion.js
/**
 * @param {string} version1
 * @param {string} version2
 * @return {number}
 * corner case: 1.3-1.2.4; 1.2-1.2.4; 01-1;
 * split two strings to arrays, and parseInt every number in case of (01, 0); then add 0 to each array to make their length are equal. Then use two pointers to iterate through.
 * Time O(n) Space O(n) 
 */
var compareVersion = function(version1, version2) {
    var arr1, arr2,
        i = 0,
        j = 0;
    arr1 = version1.split('.').map(function(x) {
        return parseInt(x) });
    arr2 = version2.split('.').map(function(x) {
        return parseInt(x) });
    var len1 = arr1.length,
        len2 = arr2.length;
    while (len1 > len2) {
        arr2.push(0);
        len1--;
    }
    while (len2 > len1) {
        arr1.push(0);
        len2--;
    }
    for (; i < arr1.length, j < arr2.length; i++, j++) {
        if (arr1[i] > arr2[j]) {
            return 1;
        } else if (arr1[i] < arr2[j]) {
            return -1;
        } else {
            continue;
        }
    }
    return 0;

};